File: core/src/ru/myitschool/cubegame/math/IntervalAction.java
package ru.myitschool.cubegame.math;

import java.util.Random;

/**
 * Created by Voyager on 05.08.2017.
 */
public class IntervalAction extends MathAction {

    private int min;
    private int max;
    private Random random;

    public IntervalAction(int min, int max) {
        this.min = min;
        this.max = max;
        random = new Random();
    }

    @Override
    public int act() {
        return random.nextInt(max + 1 - min) + min;
    }

    @Override
    public String getDescription() {
        return "(" + min + "-" + max + ")";
    }

    @Override
    public int max() {
        return max;
    }

    @Override
    public int min() {
        return min;
    }
}

File: core/src/com/wolg_vlad/dcrawler/math/IntervalAction.kt
package com.wolg_vlad.dcrawler.math

import java.util.*

/**
 * Created by Voyager on 05.08.2017.
 */
class IntervalAction(private val min: Int, private val max: Int) : MathAction() {
    private val random: Random
    override fun act(): Int {
        return random.nextInt(max + 1 - min) + min
    }

    override val description: String?
        get() = "($min-$max)"

    override fun max(): Int {
        return max
    }

    override fun min(): Int {
        return min
    }

    init {
        random = Random()
    }
}